web/src/main/java/org/hillview/BookmarkServlet.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.*;
+import org.apache.commons.io.FileUtils;
+
+@WebServlet("/bookmark")
+public class BookmarkServlet extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    public static final String bookmarkDirectory = "bookmark";
+    public static final String bookmarkExtension = ".txt";
+
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```

Comment:
This doesn't work as expected because HillviewToplevel's constructor will be initialized before index.html calling `onload()`. Specifically, HillviewToplevel constructor is initialized  when hillview.ts does `import {createHillview} from "./toplevel";`. 

I will use the HillviewToplevel to pass the bookmark filename, but not as the constructor argument. Then, I will delete the servlet because now the index.html can handle input params from the url. The bookmark will looks like this: `http://localhost:8080/?bookmark=d15d96d5-ae0b-407e-bc84-f0f2b98812ab.txt`